src/cli/build.rs

```
@@ -16,14 +25,201 @@ pub struct BuildCommand {
     #[clap(short, long)]
     offline: bool,
 
-    /// Overrides the default kernel command line
-    #[clap(short, long)]
-    kernel_cmd: Option<String>,
+    /// Overrides the default kernel image
+    #[clap(
+        short,
+        long,
+        default_value = "./build/lumper/kernel/linux-cloud-hypervisor"
+    )]
+    kernel: String,
+
+    /// Overrides the default rootfs
+    #[clap(short, long, default_value = "./build/lumper/rootfs/alpine-minirootfs")]
+    rootfs: String,
+
+    /// Overrides the default rootfs
+    #[clap(
+        short = 'K',
+        long,
+        default_value = "./build/kaps/target/x86_64-unknown-linux-musl/release/kaps"
+    )]
+    kaps: String,
+
+    /// Overrides the default bundle
+    #[clap(short, long, default_value = "./build/kaps/hack/ctr-bundle")]
+    bundle: String,
+}
+
+fn copy_dir(src: &str, workdir: &str) -> Result<()> {
+    println!("Copying directory {} to {}", src, workdir);
+    std::process::Command::new("cp")
+        .arg("-r")
+        .arg(src)
+        .arg(workdir)
+        .status()
+        .unwrap();
+
+    Ok(())
 }
 
 /// Method that will be called when the command is executed.
 impl Handler for BuildCommand {
     fn handler(&self) -> Result<()> {
+        // If offline is not set, display error message
+        let workdir = "/tmp/quark/";
+
+        if !self.offline {
+            println!("Online mode is not supported yet.");
+            return Ok(());
```

If it's not supported, you should return an error.